apps/coscrad-frontend/src/components/dynamicViews/dynamicResourceIndexView/DynamicIndexPresenter.tsx

```diff
@@ -0,0 +1,72 @@
+import { ClassDataTypeMetadata } from '@coscrad/data-types';
+import { Link } from 'react-router-dom';
+import { CommandInfo, CommandPanel } from '../commands';
+
+export type DetailDataAndActions = {
+    data: unknown[];
+    actions: unknown[];
+};
+
+export type IndexProps = {
+    schema: ClassDataTypeMetadata;
+    data: DetailDataAndActions[];
+    actions: unknown[];
+};
+
+/**
+ * TODO This should take in a `CoscradDataType` and return a formatter that maps
+ * data of said type to a readable string (presentation). JSON.stringify is
+ * being used as a catch-all. This logic should exist in a separate lib.
+ */
+const getFormatterForDataType = (_dataType: string) => (data: unknown) => JSON.stringify(data);
+
+export default ({ data, schema, actions }: IndexProps): JSX.Element => {
+    if (!Array.isArray(data))
+        return (
+            <div>
+                Error: resource index data must be an array. Received: ${JSON.stringify(data)}
+            </div>
+        );
+
+    return (
+        <div>
+            <table>
+                <thead>
```

Comment:
We can convert this using MUI Datagrid?

Reply:
Good call. That's on the TODO list- using the same approach we used in the dictionary.

Reply:
[Here](https://www.pivotaltracker.com/story/show/183456676) is a story for that work.